Assets/MixedRealityToolkit.Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -177,17 +177,17 @@ public IMixedRealitySpatialAwarenessObserver GetObserver(string name)
         }
 
         /// <inheritdoc />
-        public IMixedRealitySpatialAwarenessObserver GetObserver<T>(string name) where T : IMixedRealitySpatialAwarenessObserver
+        public T GetObserver<T>(string name) where T : IMixedRealitySpatialAwarenessObserver
         {
             for (int i = 0; i < observers.Count; i++)
             {
                 if ((observers[i] is T) && (observers[i].Name == name))
                 {
-                    return observers[i];
+                    return (T)observers[i];
                 }
             }
-
-            return null;
```

Comment:
you cannot return a null since the compiler does not know if T might be a value type. default(T) should return null for ref types, like the observers.

Reply:
Perfect, thanks!